finetuner/tuner/onnx.py

```diff
@@ -0,0 +1,158 @@
+import numpy as np
+import onnxruntime
+
+from typing import (
+    Any,
+    List,
+    Optional,
+    Tuple,
+    Union,
+)
+
+from finetuner.helper import get_framework
+
+
+def to_onnx(
+    embed_model: 'AnyDNN',
+    path: str,
+    input_shape: Union[Tuple[int], List[int]],
+    batch_size: Optional[Union[int, None]] = None,
+    opset_version: int = 11,
+) -> None:
+    """Func that converts a given model in paddle, torch or keras,
+    and converts it to the ONNX format
+    :param embed_model: Model to be converted and stored in ONNX
+    :param path: Path to store ONNX model to
+    :param input_shape: Input shape of embedding model
+    :param batch_size: The batch size the model was trained with. This
+      is only required when converting pyTorch model to onnx. In other
+      frameworks, the batch size does not need to be made specific
+    :param opset_version: ONNX opset version in which to register
+    """
+    if isinstance(input_shape, tuple):
+        input_shape = list(input_shape)
+
+    if not path.endswith(".onnx"):
+        raise ValueError(f"The `path` needs to end with `.onnx`, but was: {path}")
+
+    def _parse_and_apply_to_onnx_func(framework_name: str):
+        """Helper func to get _to_onnx_xyz func from framework name"""
+        return {
+            'torch': _to_onnx_torch,
+            'keras': _to_onnx_keras,
+            'paddle': _to_onnx_paddle,
+        }[framework_name](embed_model, path, input_shape, opset_version)
+
+    fm = get_framework(embed_model)
+
+    # Get framework-specific func to register model in ONNX
+    _parse_and_apply_to_onnx_func(fm)
```

Comment:
why not `getattr(globals())`. It's a single line

Reply:
Why not just `if/elif`

Reply:
I changed the approach using `globals`, because I like to make it more explicit. In the end, I am super agnostic here. globals, if/else, my way are all fine to me. It all kinda is the same. :) 
I will change it to if/else, since it might be the most basic and easy way of doing it. 
If you have objections, feel free to raise them. Otherwise I am going to wrap this up?!